diff --git a/BUILD.gn b/BUILD.gn
index bc75d8130b4..63b22fd6eb4 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -1424,7 +1424,7 @@ config("toolchain") {
   # Only targets in this file and its subdirs can depend on this.
   visibility = [ "./*" ]
 
-  defines = []
+  defines = [ "v8=z8", "v8_inspector=z8_inspector", "v8_crdtp=z8_crdtp", "V8_Dcheck=Z8_Dcheck", "V8_Fatal=Z8_Fatal", "cppgc=zppgc", "unibrow=znibrow" ]
   cflags = []
   ldflags = []
 
@@ -6552,6 +6552,8 @@ v8_component("v8_libbase") {
     defines = [ "BUILDING_V8_BASE_SHARED" ]
   }
 
+  defines += [ "v8=z8", "v8_inspector=z8_inspector", "v8_crdtp=z8_crdtp", "V8_Dcheck=Z8_Dcheck", "V8_Fatal=Z8_Fatal", "cppgc=zppgc", "unibrow=znibrow" ]
+
   if (is_posix || is_fuchsia) {
     sources += [
       "src/base/platform/platform-posix.cc",
