diff --git a/BUILD.gn b/BUILD.gn
index 382c7d3ed44..235ba4d9cd9 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -16,6 +16,7 @@ import("//third_party/icu/config.gni")
 
 import("gni/cluster_build.gni")
 import("gni/snapshot_toolchain.gni")
+import("gni/trickshot.gni")
 import("gni/v8.gni")
 
 if (is_clang) {
@@ -1587,6 +1588,9 @@ config("toolchain") {
   visibility = [ "./*" ]
 
   defines = []
+  if (v8_trickshot) {
+    defines += [ "v8=y8", "v8_inspector=y8_inspector", "v8_crdtp=y8_crdtp", "V8_Dcheck=Y8_Dcheck", "V8_Fatal=Y8_Fatal", "cppgc=yppgc", "heap=yeap", "unibrow=ynibrow" ]
+  }
   cflags = []
   ldflags = []
 
@@ -7235,6 +7239,10 @@ v8_component("v8_libbase") {
     defines = [ "BUILDING_V8_BASE_SHARED" ]
   }
 
+  if (v8_trickshot) {
+    defines += [ "v8=y8", "v8_inspector=y8_inspector", "v8_crdtp=y8_crdtp", "V8_Dcheck=Y8_Dcheck", "V8_Fatal=Y8_Fatal", "cppgc=yppgc", "heap=yeap", "unibrow=ynibrow" ]
+  }
+
   if (is_posix || is_fuchsia) {
     sources += [
       "src/base/platform/platform-posix.cc",
diff --git a/gni/trickshot.gni b/gni/trickshot.gni
new file mode 100644
index 00000000000..af6d5f9f385
--- /dev/null
+++ b/gni/trickshot.gni
@@ -0,0 +1,4 @@
+declare_args() {
+  # Macro to replace 'v8' with 'y8' for shared library linkage
+  v8_trickshot = false
+}
