diff --git a/gni/v8.gni b/gni/v8.gni
index 44f056cf..05acb1cf 100644
--- a/gni/v8.gni
+++ b/gni/v8.gni
@@ -375,7 +375,7 @@ if (v8_symbol_level != symbol_level) {
   }
 }
 
-if ((is_posix || is_fuchsia) && (v8_enable_backtrace || v8_monolithic)) {
+if ((is_posix || is_fuchsia) && v8_enable_backtrace) {
   v8_remove_configs += [ "//build/config/gcc:symbol_visibility_hidden" ]
   v8_add_configs += [ "//build/config/gcc:symbol_visibility_default" ]
 }
