diff --git a/src/runtime/runtime-test.cc b/src/runtime/runtime-test.cc
index 12a2ead2d07..00ff940e335 100644
--- a/src/runtime/runtime-test.cc
+++ b/src/runtime/runtime-test.cc
@@ -1173,7 +1173,7 @@ RUNTIME_FUNCTION(Runtime_SetAllocationTimeout) {
   CONVERT_INT32_ARG_FUZZ_SAFE(timeout, 1);
   isolate->heap()->set_allocation_timeout(timeout);
 #else   // !V8_ENABLE_ALLOCATION_TIMEOUT
-  static std::atomic_flag printed_warning{false};
+  static std::atomic_flag printed_warning;
   if (!printed_warning.test_and_set()) {
     base::OS::PrintError(
         "Warning: %%SetAllocationTimeout has no effect in this build. Set the "
